haproxy/userdata.sh.tpl

```diff
@@ -5,6 +5,18 @@ apt-get -y install haproxy
 
 tee -a /etc/haproxy/haproxy.cfg <<EOF
 
+resolvers awsdns
+  nameserver aws ${dns_ip}:53
+  resolve_retries       10
+  timeout resolve       2s
+  timeout retry         1s
+  hold other           30s
+  hold refused         30s
+  hold nx              30s
+  hold timeout         30s
+  hold valid           30s
+  hold obsolete        30s
```

Comment:
I still think we shouldn't veer from the default, especially looking at `obsolete`. If I understand the docs correctly it seems that when an ELB changes IP addresses, the old one will still be valid for an extra 30secs and can potentially cause issues again? (although in my experience, even after the AWS nameserver stops responding with an IP address it will still be valid at least for a few minutes)